apps/condo/domains/user/schema/RegisterNewUserThroughExternalIdentityService.js

```diff
@@ -0,0 +1,176 @@
+    },
+    WRONG_SENDER_FORMAT: {
+        mutation: 'registerNewUserThroughExternalIdentity',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value. {details}',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
+    IDENTITY_INTEGRATION_DOES_NOT_SUPPORT_LOGIN_FORM: {
+        mutation: 'registerNewUserThroughExternalIdentity',
+        variable: ['data', 'identityType'],
+        code: BAD_USER_INPUT,
+        type: IDENTITY_INTEGRATION_DOES_NOT_SUPPORT_LOGIN_FORM,
+        message: 'Identity integration does not support login form',
+    },
+    EXTERNAL_IDENTITY_ALREADY_REGISTERED_ERROR: {
+        mutation: 'registerNewUserThroughExternalIdentity',
+        variable: ['data', 'tokenSet'],
+        code: BAD_USER_INPUT,
+        type: EXTERNAL_IDENTITY_ALREADY_REGISTERED_ERROR,
+        message: 'External identity already registered',
+        messageForUser: 'api.user.registerNewUserThroughExternalIdentity.EXTERNAL_IDENTITY_ALREADY_REGISTERED_ERROR',
+    },
+    USER_WITH_SPECIFIED_PHONE_ALREADY_EXISTS: {
+        mutation: 'registerNewUserThroughExternalIdentity',
+        code: BAD_USER_INPUT,
+        type: NOT_UNIQUE,
+        message: 'User with specified phone already exists',
+        messageForUser: 'api.user.registerNewUser.USER_WITH_SPECIFIED_PHONE_ALREADY_EXISTS',
+    },
+    USER_WITH_SPECIFIED_EMAIL_ALREADY_EXISTS: {
+        mutation: 'registerNewUserThroughExternalIdentity',
+        code: BAD_USER_INPUT,
+        type: NOT_UNIQUE,
+        message: 'User with specified email already exists',
+        messageForUser: 'api.user.registerNewUser.USER_WITH_SPECIFIED_EMAIL_ALREADY_EXISTS',
+    },
+}
+
+async function ensureNotExists (context, field, value) {
+    const existed = await User.getOne(context, { [field]: value, type: RESIDENT })
+    if (existed) {
+        throw new GQLError({
+            phone: errors.USER_WITH_SPECIFIED_PHONE_ALREADY_EXISTS,
+            email: errors.USER_WITH_SPECIFIED_EMAIL_ALREADY_EXISTS,
+        }[field], context)
+    }
+}
+
+const RegisterNewUserThroughExternalIdentityService = new GQLCustomSchema('RegisterNewUserThroughExternalIdentityService', {
+    types: [
+        {
+            access: true,
+            type: 'input RegisterNewUserThroughExternalIdentityInput { dv: Int! sender: SenderFieldInput!, identityType: IdentityType!, tokenSet: JSON!, name: String! }',
+        },
+        {
+            access: true,
+            type: 'type RegisterNewUserThroughExternalIdentityOutput { item: User!, token: String! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canRegisterNewUserThroughExternalIdentity,
+            schema: 'registerNewUserThroughExternalIdentity(data: RegisterNewUserThroughExternalIdentityInput!): RegisterNewUserThroughExternalIdentityOutput',
```

Comment:
it looks like server-side callback logic

Reply:
migrated